File: 11/02/tests/login_test.py
import pytest
from pages import login_page


class TestLogin():

    @pytest.fixture
    def login(self, driver):
        return login_page.LoginPage(driver)

    def test_valid_credentials(self, login):
        login.with_("tomsmith", "SuperSecretPassword!")
        assert(login.success_message_present())

    def test_invalid_credentials(self, login):
        login.with_("tomsmith", "bad password")
        assert(login.failure_message_present())
        # assert(self.login.success_message_present() == False)

File: code-examples/javascript/10/01/test/LoginTest.js
const { Builder } = require('selenium-webdriver')
const path = require('path')
const assert = require('assert')
const LoginPage = require('../pages/LoginPage')

describe('Login', function() {
  this.timeout(30000)
  let driver
  let login

  beforeEach(async function() {
    const vendorDirectory =
      path.delimiter + path.join(__dirname, '..', 'vendor')
    process.env.PATH += vendorDirectory
    driver = await new Builder().forBrowser('firefox').build()
    login = new LoginPage(driver)
    await login.load()
  })

  afterEach(async function() {
    await driver.quit()
  })

  it('with valid credentials', async function() {
    await login.authenticate('tomsmith', 'SuperSecretPassword!')
    assert(await login.successMessagePresent(), 'Success message not displayed')
  })

  it('with invalid credentials', async function() {
    await login.authenticate('tomsmith', 'bad password')
    assert(await login.failureMessagePresent(), 'Failure message not displayed')
  })
})
